### PR TITLE
Solved Issues #3 #5 #7

### PR DESCRIPTION
**Pull request by NeonKazuha**
1. Implemented Text Encoding Techniques to convert Categorical Data into Numeric Data using technique of Label Encoding,
- The features 'region', 'type', 'wine' have been Label Encoded using the LabelEncoder from preprocessing module in the sklearn library. 

Screenshot 2024-03-15 034351

2. Created Correlation and Confusion matrix.
Screenshot 2024-03-15 041523

3. Various Visualization Methods.
- Multiple different graphs such as Violin Plots, Box Plots and Scatter Plots have been created to provide a more clear understanding of Data.
Screenshot 2024-03-15 041608
Screenshot 2024-03-15 041627
Screenshot 2024-03-15 041646

4. Multiple Different Models.
- Multiple models such as Decision Tree Classifier, SVC, Random Forest Classifier, etc. used in a loop to automate a dictionary of various models, fit training data and find the accuracy score of each model on the testing set . The model with the Best accuracy is selected to predict future values.
Screenshot 2024-03-15 041809

### CODE REVIEW

**Comment by NeonKazuha** [comment]
@darshbaxi @rootCircle Please Review.

**Comment by darshbaxi** [comment]
I've closed this PR because it addressed multiple issues simultaneously. For better tracking and clarity, each issue should ideally have its own separate PR.
